### DISCUSSION TITLE
Yggdrasil and no SSU2 connections

### DISCUSSION THREAD

**Discussion by bjtftw** [PurpleI2P/i2pd#1985] [General]
Why i2pd doesn't use SSU2 when in MESH mode (ipv4=0, ipv6=0, yggdrasil=1, [ssu2] and [ntcp2] both enabled=1)?

When I look at 127.0.0.1:7070 Transports there is no single UDP/SSU connection?

When I enable additionally one of ipv4 or ipv6 with yggdrasil enabled only those protocols use SSU2 but still no single SSU2 connection from yggdrasil IPs. It looks like it's intentionally disabled somewhere in the i2pd code for yggdrasil connections, so what's the reason for it?

**Comment by Vort** [comment]
I think SSU2 over Yggdrasil is not needed.
With real network UDP packets can be lost.
While Yggdrasil connections are reliable and therefore more close to TCP than to UDP.
